services/tenant-ui/frontend/src/store/verifierStore.ts

```diff
@@ -21,11 +21,17 @@ export const useVerifierStore = defineStore('verifier', () => {
     return fetchList('/tenant/v1/verifier/presentations/', presentations, error, loading);
   }
 
-  async function getPresentationDetails(verifier_presentation_id: string) {
-    return fetchItemDetailtoCache('/tenant/v1/verifier/presentations/', verifier_presentation_id, presentationDetailCache, error, loading, { acapy: true });
-  }
-
-  return { presentations, presentationDetailCache, selectedPresentation, loading, error, listPresentations, getPresentationDetails };
+  const presentationDetailbyId = computed((verifier_presentation_id: string, forceFetch: boolean = false) => {
+    console.log(verifier_presentation_id)
+    let item = presentationDetailCache[verifier_presentation_id];
+    if (!item || forceFetch) {
+      item = fetchItem('/tenant/v1/verifier/presentations/', verifier_presentation_id, error, loading, { acapy: true });
+      presentationDetailCache[verifier_presentation_id] = item;
+    }
+    return item
+  })
+
+  return { presentations, presentationDetailCache, selectedPresentation, loading, error, listPresentations, presentationDetailbyId };
```

Comment:
since you are trying to establish a good pattern for future developers, let's just finish this off. 

1. don't expose the `presentationDetailCache`
2. add in a true API call to getPresentationDetail, that does not use cache and allows callers to ask for whatever shape they need (timeline, acapy, deleted...) this will use the fetchItem.
3. naming that getter `presentationDetailById` is not very clear, so let's make a distinction between the "true" API call and this `getPresentationDetails` and `getPresentationDetailsFromCache`
4. the `forceFetch` is great, but we should really be checking the `presentations` too for staleness.  like if we know we have fetched a more recent item then we know that cache is stale. it may not be 100% because we rely on that list being refreshed but it's there and we can do an additional safety check.
5. maybe not this PR, but putting this cache idea into its own store class will allow us to isolate the loading flag to loading the cache as opposed to the main collection.